### ISSUE TITLE
Maybe codec doesn't work for non-existent keys

### ISSUE THREAD

**Issue by chenguo** [gigobyte/purify#189]
Simple example:

```
const MyCodec = Codec.interface({
  foo: maybe(Codecs.number),
  bar: string,
});
const result = MyCodec.decode({ bar: 'baz'});
```

I got this:
```
{
  __value: 'Problem with property "foo": it does not exist in received object {"bar":"baz"}'
}
```

By the nature of `Maybe` I would've expected a `Nothing` value for `foo` here rather than a decode fail. If I decode `{ foo: undefined, bar: 'baz'})` I get the expected result of a `Nothing` for `foo`.

**Comment by chenguo** [comment]
I can work around it with this for now:

```
const maybe = <T>(codec: Codec<T>) => Codecs.optional(Codecs.maybe(codec));
```

**Comment by gigobyte** [comment]
This is fixed on master and available in version `0.16.0-beta.1`. You can either install that or use the workaround, both are fine.

**Comment by chenguo** [comment]
That didn't quite work for me, I think when you passed `_isOptional` through `custom`, the `custom` codec drops it. I PR-ed #190 that should work, lemme know what you think?

**Comment by chenguo** [comment]
Thanks for reviewing & merging this so quickly! Can we maybe get a release pushed up to NPM? That would save me the trouble of uploading a dev version to our internal NPM  server.

Thanks for the work on this project too. I tried out `fp-ts` and it just wasn't to my liking, having come from a Scala environment I enjoy your interface decisions much more than those in `fp-ts`.

**Comment by gigobyte** [comment]
@chenguo, Sure, I've published `0.16.0-beta.2`
